cmd/query/app/http_handler.go

```diff
@@ -78,23 +78,18 @@ func NewRouter() *mux.Router {
 
 // APIHandler implements the query service public API by registering routes at httpPrefix
 type APIHandler struct {
-	spanReader        spanstore.Reader
-	archiveSpanReader spanstore.Reader
-	archiveSpanWriter spanstore.Writer
-	dependencyReader  dependencystore.Reader
-	adjuster          adjuster.Adjuster
-	logger            *zap.Logger
-	queryParser       queryParser
-	basePath          string
-	apiPrefix         string
-	tracer            opentracing.Tracer
+	queryService *querysvc.QueryService
+	queryParser  queryParser
+	basePath     string
+	apiPrefix    string
+	logger       *zap.Logger
+	tracer       opentracing.Tracer
 }
 
 // NewAPIHandler returns an APIHandler
 func NewAPIHandler(spanReader spanstore.Reader, dependencyReader dependencystore.Reader, options ...HandlerOption) *APIHandler {
```

Comment:
I suggest accepting QueryService instead of creating it internally, otherwise you don't have a way to apply its additional configuration options.